akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ReplayingSnapshot.scala

```diff
@@ -168,6 +169,11 @@ private[akka] class ReplayingSnapshot[C, E, S](override val setup: BehaviorSetup
       }
 
       setup.internalLogger.debug("Snapshot recovered from {} {} {}", seqNr, seenPerReplica, version)
+      snapshot.foreach { snap =>
+        import akka.persistence.typed.{ SnapshotMetadata => TypedSnapshotMetadata }
+
+        setup.onSignal(state, SnapshotRecovered(TypedSnapshotMetadata.fromClassic(snap.metadata)), catchAndLog = true)
+      }
```

Comment:
maybe move a few lines below, after `setup.currentMetadata = metadata`, so that currentSequenceNumber is set when emitting the signal